pkg/kubelet/util/manager/watch_based_manager.go

```diff
@@ -260,17 +261,20 @@ func (c *objectCache) AddReference(namespace, name string) {
 		item = c.newReflectorLocked(namespace, name)
 		c.items[key] = item
 	}
-	item.refCount++
+	item.refMap[podUID]++
 }
 
-func (c *objectCache) DeleteReference(namespace, name string) {
+func (c *objectCache) DeleteReference(namespace, name string, podUID types.UID) {
 	key := objectKey{namespace: namespace, name: name}
 
 	c.lock.Lock()
 	defer c.lock.Unlock()
 	if item, ok := c.items[key]; ok {
-		item.refCount--
-		if item.refCount == 0 {
+		item.refMap[podUID]--
+		if item.refMap[podUID] == 0 {
+			delete(item.refMap, podUID)
+		}
+		if len(item.refMap) == 0 {
```

Comment:
Can you create a more targeted unit test for this logic?

Reply:
I added some testcases, is it to much or are there any missing scenarios?